src/util/simplify_expr.cpp

```diff
@@ -361,11 +359,38 @@ static simplify_exprt::resultt<> simplify_string_index_of(
 
     const array_exprt &s2_data = s2_data_opt->get();
 
-    it = std::search(
-      std::next(s1_data.operands().begin(), starting_index),
-      s1_data.operands().end(),
-      s2_data.operands().begin(),
-      s2_data.operands().end());
+    if(search_from_end)
+    {
+      auto starting_offset = 0;
+      if(starting_index > 0)
+        starting_offset = search_string_size - starting_index;
```

Comment:
Use ternary if to define starting_offset and make it const. Avoid using `auto` when the type is not explicit.